src/sagemaker/jumpstart/curated_hub/curated_hub.py

```diff
@@ -13,37 +13,97 @@
 """This module provides the JumpStart Curated Hub class."""
 from __future__ import absolute_import
 
-from typing import Optional, Dict, Any
-
+from typing import Any, Dict, Optional
+import boto3
 from sagemaker.session import Session
+from sagemaker.jumpstart.constants import (
+    JUMPSTART_DEFAULT_REGION_NAME,
+)
+
+from sagemaker.jumpstart.types import HubDescription, HubContentType, HubContentDescription
+import sagemaker.jumpstart.session_utils as session_utils
 
 
 class CuratedHub:
     """Class for creating and managing a curated JumpStart hub"""
 
-    def __init__(self, hub_name: str, region: str, session: Optional[Session] = None):
+    def __init__(
+        self,
+        hub_name: str,
+        region: str = JUMPSTART_DEFAULT_REGION_NAME,
+        sagemaker_session: Optional[Session] = None,
```

Comment:
We'll have conflicts in my PR #4439, but we should default to `DEFAULT_JUMPSTART_SAGEMAKER_SESSION` here

Reply:
I actually thought about that and decided to derive session from the region if left None.`DEFAULT_JUMPSTART_SAGEMAKER_SESSION` is basically Session with `JUMPSTART_DEFAULT_REGION_NAME` which is `us-west-2`. It is a bit weird to get both region and the session, even with the regions check in line 37. I guess we can derive the region from the sagemaker_session actually.

Reply:
Sorry just seeing this reply- yes you're totally right. We don't need the `region` param since we can derive that from `session`